### ISSUE TITLE
Docker vcpkg failed to rename file

### ISSUE THREAD

**Issue by GrandChris** [microsoft/vcpkg#14098]
**Describe the bug**
I read about "https://code.visualstudio.com/docs/remote/containers" and I wanted to try it with vcpkg. 
I'm using Ubuntu on Docker on Windows. Vcpkg is used as a git submodule. When I try to install a package I get an error message with "failed to rename file" and the installation stops. I've tried it with "sqlite3" and "gtest", both with the same result.

**Environment**
- OS:  Windows Version	10.0.18363 Build 18363
- Docker:
```DockerFile
FROM ubuntu:groovy

RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
    && apt-get -y install --no-install-recommends --reinstall ca-certificates \
    && apt-get -y install --no-install-recommends git build-essential cmake ninja-build curl unzip tar
```
- Compiler: gcc (Ubuntu 10.2.0-13ubuntu1) 10.2.0
- vcpkg: 2020.06.15-unknownhash
  Using vcpkg branch mastter, Hash: 27a2418e91179d8607862348d7b498558e98a0ab

**To Reproduce**
Use docker container

1. ./vcpkg/bootstrap-vcpkg.sh
2. ./vcpkg/vcpkg install sqlite3

**Expected behavior**
Build library without failure.

**Failure logs** 
```
root@31f7a9c74a9c:/workspaces/ucppcoro/vcpkg# ./vcpkg install sqlite3
Warning: manifest-root detected at /workspaces/ucppcoro, but manifests are not enabled.
If you wish to use manifest mode, you may do one of the following:
    * Add the `manifests` feature flag to the comma-separated environment
      variable `VCPKG_FEATURE_FLAGS`.
    * Add the `manifests` feature flag to the `--feature-flags` option.
    * Pass your manifest directory to the `--x-manifest-root` option.
If you wish to silence this error and use classic mode, you can:
    * Add the `-manifests` feature flag to `VCPKG_FEATURE_FLAGS`.
    * Add the `-manifests` feature flag to `--feature-flags`.
Computing installation plan...
The following packages will be built and installed:
    sqlite3[core]:x64-linux
Detecting compiler hash for triplet x64-linux...
Starting package 1/1: sqlite3:x64-linux
Building package sqlite3[core]:x64-linux...
Could not locate cached archive: /root/.cache/vcpkg/archives/b0/b003c9ccfe189ff994ca4680caf4e46af65fe6fe.zip
-- Downloading https://sqlite.org/2020/sqlite-amalgamation-3330000.zip...
-- Extracting source /workspaces/ucppcoro/vcpkg/downloads/sqlite-amalgamation-3330000.zip
-- Applying patch fix-arm-uwp.patch
CMake Error at scripts/cmake/vcpkg_extract_source_archive_ex.cmake:147 (file):
  file RENAME failed to rename

    /workspaces/ucppcoro/vcpkg/buildtrees/sqlite3/src/3330000-c6f2129a7a.tmp/sqlite-amalgamation-3330000

  to

    /workspaces/ucppcoro/vcpkg/buildtrees/sqlite3/src/3330000-c6f2129a7a.clean

  because: No such file or directory

Call Stack (most recent call first):
  ports/sqlite3/portfile.cmake:10 (vcpkg_extract_source_archive_ex)
  scripts/ports.cmake:79 (include)


Error: Building package sqlite3:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: sqlite3:x64-linux
  Vcpkg version: 2020.06.15-unknownhash

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
sqlite3 folder structure
```
root@31f7a9c74a9c:/workspaces/ucppcoro/vcpkg/buildtrees/sqlite3# tree
.
|-- extract-err.log
|-- extract-out.log
|-- patch-x64-linux-0-err.log
|-- patch-x64-linux-0-out.log
|-- src
|   `-- 3330000-c6f2129a7a.tmp
|       |-- sqlite-amalgamation-3330000
|       |   |-- shell.c
|       |   |-- sqlite3.c
|       |   |-- sqlite3.h
|       |   `-- sqlite3ext.h
|       `-- sqlite-amalgamation-3330000.zip.extracted
|-- stdout-x64-linux.log
`-- x64-linux.vcpkg_abi_info.txt
```

**Comment by GrandChris** [comment]
Starting my PC today, Docker asked me to use WSL2 instead of WSL1. I switched to WSL2 and now sqlite builds without failure. So the encountered bug depends on WSL1 and if it is for me, the issue can be closed.

**Comment by BillyONeal** [comment]
Do you have some sort of antivirus program on this system? WSL1 would expose that file to the AV tool but WSL2 would not.

**Comment by GrandChris** [comment]
Nothing special, only Windows Defender I guess.

**Comment by BillyONeal** [comment]
Closing this as a duplicate of https://github.com/microsoft/vcpkg/issues/8199
